src/mui/input/ReferenceArrayInput.js

```diff
@@ -172,6 +176,43 @@ export class ReferenceArrayInput extends Component {
             );
         }
 
+        if (matchingReferences === null) {
+            return (
+                <Labeled
+                    label={
+                        typeof label === 'undefined' ? (
+                            `resources.${resource}.fields.${source}`
+                        ) : (
+                            label
+                        )
+                    }
+                >
+                    <span style={progessContainerStyle}>
+                        <LinearProgress
+                            mode="indeterminate"
+                            style={progessStyle}
+                        />
+                    </span>
+                </Labeled>
+            );
+        }
+
+        if (matchingReferences.error) {
+            return (
+                <TextField
+                    disabled={true}
+                    hintText={
+                        typeof label === 'undefined' ? (
+                            `resources.${resource}.fields.${source}`
+                        ) : (
+                            label
+                        )
+                    }
+                    errorText={matchingReferences.error}
+                />
+            );
+        }
+
         if (!(referenceRecords && referenceRecords.length > 0) && !allowEmpty) {
```

Comment:
IMO, `refrenceRecords` should be deprecated. @fzaninotto @djhi are you agree with that ?

Reply:
Can you elaborate ?

Reply:
Here, we test if we have the referenceIds data in the state resources
```
const referenceIds = props.input.value || [];
const data = state.admin.resources[props.reference].data;
referenceRecords: referenceIds.reduce((references, referenceId) => {
            if (data[referenceId]) {
                references.push(data[referenceId]);
            }
            return references;
        }, []),
```
If not, and if `allowEmpty` is not `true`, we show a disabled input. 
I think we don't need to do this anymore (because we test if value is required with `validate`, and not with `allowEmpty` prop), so we should remove this `recordReference` from props.